homework/list.js

```diff
@@ -1,79 +1,100 @@
 'use strict';
 
-(function() {
+var ListManager = (function() {
+  
+  // constructor
+  var self = function() {
+    this._listNoteContent = [];
+    this._wrapper = null;
+  };
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+  // prototype
+  self.prototype = {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+    start() {
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+      this._wrapper = document.querySelector('#note-list-wrapper');
+      this.fetchList()
+      .then(this.handler.bind(this));
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+      window.addEventListener('click', (function(event) {
```

Comment:
It's a convention to prefix method with `on` for the corresponding function, so it should be renamed as 'onClick'. And, to set `this` as the listener is better than an anonymous function.